src/component/user/DemoAuth/DemoAuth.jsx

```diff
@@ -1,25 +1,24 @@
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import { Button, TextField } from '@material-ui/core';
-import useUser from '../../../hooks/api/getters/useUser/useUser';
-
+import { useAuth } from '../../../hooks/api/getters/useAuth/useAuth';
 import styles from './DemoAuth.module.scss';
-
 import { ReactComponent as Logo } from '../../../assets/img/logo.svg';
 import { LOGIN_BUTTON, LOGIN_EMAIL_ID } from '../../../testIds';
+import { useHistory } from 'react-router-dom';
+import { useAuthApi } from '../../../hooks/api/actions/useAuthApi/useAuthApi';
 
-const DemoAuth = ({ demoLogin, history, authDetails }) => {
+const DemoAuth = ({ authDetails }) => {
     const [email, setEmail] = useState('');
-
-    const { refetch } = useUser();
+    const history = useHistory();
+    const { refetchAuth } = useAuth();
+    const { emailAuth } = useAuthApi();
 
     const handleSubmit = evt => {
         evt.preventDefault();
-        const user = { email };
-        const path = evt.target.action;
 
-        demoLogin(path, user).then(() => {
-            refetch();
+        emailAuth(authDetails.path, email).then(() => {
```

Comment:
I think we should try to land a pattern here too. Most other parts use try / catch.